Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb

```diff
@@ -207,46 +207,46 @@ def process(queue)
     end # begin
   end # process
   
-  # Get first json object out of a string, return the rest of the string
-  def get_jsons(content, batch_size)
-    return nil, content, 0 if content.nil? || content.length == 0
-    return nil, content, 0 if (content.index '{').nil?
-
-    hit = 0
-    count = 0
-    index = 0
-    first = content.index('{')
-    move_opening = true
-    move_closing = true
-    while(hit < batch_size)
-        inIndex = content.index('{', index) if move_opening
-        outIndex = content.index('}', index) if move_closing
-
-        # TODO: Fix the ending condition
-        break if count == 0 && (inIndex.nil? || outIndex.nil?)
-        
-        if(inIndex.nil?)
-            index = outIndex
-        elsif(outIndex.nil?)
-            index = inIndex
-        else
-            index = [inIndex, outIndex].min
-        end #if
-        if content[index] == '{'
-            count += 1
-            move_opening = true
-            move_closing = false
-        elsif content[index] == '}'
-            count -= 1
-            move_closing = true
-            move_opening = false
-        end #if
-        index += 1
-        hit += 1 if count == 0
-    end
+# Get json objects out of a string and return it. Note, content will be updated in place as well.
+def get_jsons!(content, batch_size)
+  return nil if content.nil? || content.length == 0
+  return nil if (content.index '{').nil?
+
+  hit = 0
+  count = 0
+  index = 0
+  first = content.index('{')
+  move_opening = true
+  move_closing = true
+  while(hit < batch_size)
+    inIndex = content.index('{', index) if move_opening
```

Comment:
don't forget to exclude '{' and '}' inside string values

Reply:
Good feedback. That scenario is tracked by #85. I prefer to deal with it separately.